README

	     smtm --- A global stock ticker for X11 and Windoze


1. Introduction
   
   smtm, which is a not overly clever acronym for Show Me The Money, is a
   financial ticker and portfolio application for quotes from exchanges
   around the world (provided they are carried on Yahoo!). It creates and 
   automatically updates a window with quotes from Yahoo! Finance. It can 
   also display the entire variety of charts available at Yahoo! Finance. 
   When called with one or several symbols, it displays these selected
   stocks. When smtm is called without arguments, it reads the symbols
   tickers from a file, by default ~/.smtmrc. This file can be created
   explicitly by calling the Save option from the File menu. Beyond stocks,
   smtm can also display currencies (from the Philadephia exchange), US
   mutual funds, options on US stocks, several precious metals and quite
   possibly more; see the Yahoo! Finance website for full information. 


2. How to get started
   
   Built using Perl/Tk it requires at least the following modules
   
   	  Date::Manip
	  Finance::YahooQuote
	  LWP::UserAgent
	  MIME::Base64
	  HTML::Parser
	  Tk	
          Tk::PNG

   all of which are readily available for most Unix variants, Linux
   distributions and the ActiveState Perl implementation common on win*.
   
   Debian users just say 'apt-get install smtm'. 

   For others, and following Perl convention, the installation should work as

         perl Makefile.PL # optionally "perl Makefile.PL verbose"
         make
	 make test	  # the tests are somewhat limited	 
         make install     

   but I do not test this regularly as I install the Debian packages myself.
   

3. How to use it

   'smtm --help' gives a brief synopis, 'perldoc smtm' shows the full manual 
   page.  Several options are available and can be set from both a resource
   file ~/.smtmrc, the command-line and the menus once smtm is running.

   Additional resource files, e.g. for additional portfolios, are available
   in the examples/ directory of the source distribution.

   But the simplest case is possibly to just supply ticker symbols on the 
   command-line:  'smtm DELL IBM MSFT'.


4. Disclaimer (taken straight from the GPL)

				 NO WARRANTY

     11. BECAUSE THE PROGRAM IS LICENSED FREE OF CHARGE, THERE IS NO WARRANTY
   FOR THE PROGRAM, TO THE EXTENT PERMITTED BY APPLICABLE LAW.  EXCEPT WHEN
   OTHERWISE STATED IN WRITING THE COPYRIGHT HOLDERS AND/OR OTHER PARTIES
   PROVIDE THE PROGRAM "AS IS" WITHOUT WARRANTY OF ANY KIND, EITHER EXPRESSED
   OR IMPLIED, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
   MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.  THE ENTIRE RISK AS
   TO THE QUALITY AND PERFORMANCE OF THE PROGRAM IS WITH YOU.  SHOULD THE
   PROGRAM PROVE DEFECTIVE, YOU ASSUME THE COST OF ALL NECESSARY SERVICING,
   REPAIR OR CORRECTION.

     12. IN NO EVENT UNLESS REQUIRED BY APPLICABLE LAW OR AGREED TO IN WRITING
   WILL ANY COPYRIGHT HOLDER, OR ANY OTHER PARTY WHO MAY MODIFY AND/OR
   REDISTRIBUTE THE PROGRAM AS PERMITTED ABOVE, BE LIABLE TO YOU FOR DAMAGES,
   INCLUDING ANY GENERAL, SPECIAL, INCIDENTAL OR CONSEQUENTIAL DAMAGES ARISING
   OUT OF THE USE OR INABILITY TO USE THE PROGRAM (INCLUDING BUT NOT LIMITED
   TO LOSS OF DATA OR DATA BEING RENDERED INACCURATE OR LOSSES SUSTAINED BY
   YOU OR THIRD PARTIES OR A FAILURE OF THE PROGRAM TO OPERATE WITH ANY OTHER
   PROGRAMS), EVEN IF SUCH HOLDER OR OTHER PARTY HAS BEEN ADVISED OF THE
   POSSIBILITY OF SUCH DAMAGES.


   In short: if it breaks, you can keep the pieces.  This is Free Software,
   use it as you see fit, but don't come running or screaming if it doesn't
   work as intended. 


5. Copyright

   Copyright (C) 1999 - 2007  Dirk Eddelbuettel <edd@debian.org>

   This program is free software; you can redistribute it and/or modify
   it under the terms of the GNU General Public License as published by
   the Free Software Foundation; either version 2 of the License, or
   (at your option) any later version.

   This program is distributed in the hope that it will be useful,
   but WITHOUT ANY WARRANTY; without even the implied warranty of
   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
   GNU General Public License for more details.

   You should have received a copy of the GNU General Public License
   along with this program; if not, write to the Free Software
   Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.


debian/changelog
smtm (1.6.11) unstable; urgency=low

  * debian/rules: Use DESTDIR for required MakeMaker change in Perl 5.22
							(Closes: #795680)

  * debian/rules: Add build-arch and build-indep targets [lintian]
  * debian/control: Add Depends: on ${misc:Depends} [lintian]
  * debian/control: Update Standards-Version: to 3.9.6 [lintian]
  * debian/copyright: Refer to version'ed GPL [lintian]
  
 -- Dirk Eddelbuettel <edd@debian.org>  Sun, 16 Aug 2015 08:47:25 -0500

smtm (1.6.10) unstable; urgency=low

  * Bug fix release to accommodate Perl 5.10 array changes in two spots

 -- Dirk Eddelbuettel <edd@debian.org>  Sun, 25 May 2008 17:48:23 -0500

smtm (1.6.9.1) unstable; urgency=low

  * Debian bug fix release:
    - debian/rules: Remove bashism			(Closes: #477620)
    - debian/control: Updated Standards-Version to 3.7.3
    - debian/copyright: Added parseable Copyright statement
  
  * smtm.pl: Update copyright to 2008
  
 -- Dirk Eddelbuettel <edd@debian.org>  Wed, 23 Apr 2008 21:33:40 -0500

smtm (1.6.9) unstable; urgency=low

  * Bug fix / update release:  Updated chart URL at Yahoo!
  
  * Debian updates:
    - Standards-Version: increased to 3.7.2
    - debian/compat created, debhelper Build-Depends updated

 -- Dirk Eddelbuettel <edd@debian.org>  Mon, 10 Sep 2007 21:30:51 -0500

smtm (1.6.8) unstable; urgency=low

  * Really apply the change to Depends: and not only Build-Depends:
  * debian/control: Remove Depends: on libtk-png-perl, but qualify Depends:
    on perl-tk to be perl-tk (>= 1:804.000)

 -- Dirk Eddelbuettel <edd@debian.org>  Fri, 20 Jan 2006 21:39:36 -0600

smtm (1.6.7) unstable; urgency=low

  * Debian-only bug fix release: We no longer need to Depends: libtk-png-perl
    as newer perl-tk packages include the PNG graph functionality
  * debian/control: Remove Depends: on libtk-png-perl, but qualify Depends:
    on perl-tk to be perl-tk (>= 1:804.000)

  * debian/control: Upgraded to Standards-Version: 3.6.2
  * debian/rules: Set DH_COMPAT=4

 -- Dirk Eddelbuettel <edd@debian.org>  Fri, 20 Jan 2006 20:54:51 -0600

smtm (1.6.6) unstable; urgency=low

  * Bug fix release:
  
    - [smtm] Switch to using Tk::PNG to display png charts as Yahoo no longer
      ships gif files, and Tk by default cannot deal with PNG
 
    - [smtm] Updated documentation a little bit
  
    - [smtm] Use DBK.DE as symbol in the example portfolio
  
    - [Makefile.PL, README] Mention the new Tk::PNG requirement
  
    - [debian/control] Add libtk-png-perl to Depends

 -- Dirk Eddelbuettel <edd@debian.org>  Sun, 15 May 2005 22:34:33 -0500

smtm (1.6.5) unstable; urgency=low

  * Bug fix release:
  
    - [smtm] Make chart updates in gallery mode more robust to the possibility
      of windows having been closed by the user (Closes: #261806)

    - [smtm] Make code yet more robust to irregular data from Yahoo!
  
 -- Dirk Eddelbuettel <edd@debian.org>  Tue,  3 Aug 2004 21:52:29 -0500

smtm (1.6.4) unstable; urgency=low

  * Debian-only bug fix release: Needed explicit Depends on libdate-manip-perl

 -- Dirk Eddelbuettel <edd@debian.org>  Fri,  4 Jun 2004 07:50:42 -0500

smtm (1.6.3) unstable; urgency=low

  * Bug fix release:

    - [smtm] Fix main window resizing bug following 'edit_stock()' is called
      (via the middle-mouse button).  (Closes: #241777)
  
 -- Dirk Eddelbuettel <edd@debian.org>  Sat,  3 Apr 2004 10:18:08 -0600

smtm (1.6.2) unstable; urgency=low

  * Minor release with new feature

    - The 'show_details' function (bound to Button-1) now displays absolute
      profit or loss (converted to local currency as well). Thanks to Rober
      A. Schmied for the patch to which we only added the FX bit.
  
   -  Percentage returns will not be annualized for positions held less 
      than year, also due to Robert A. Schmied.
  
  * Bug fixes
  
    - Gallery mode works again after the innocent-looking call resizable(0,0) 
      has been removed in the view_image function.
  
    - Resizable(0,0) calls added in other functions. 

 -- Dirk Eddelbuettel <edd@debian.org>  Wed, 17 Mar 2004 22:06:00 -0600

smtm (1.6.1) unstable; urgency=low

  * Bug fix release:
    
    - The 'show_details' function (bound to Button-1) now displays correct
      information on dividend date, amount and yields. Display of market
      capitalization was also improved. That to Robert A. Schmied for 
      pointing out the initial problem, and for supplying a patch.

 -- Dirk Eddelbuettel <edd@debian.org>  Tue,  9 Mar 2004 21:15:35 -0600

smtm (1.6.0) unstable; urgency=low

  * New major release:
  
    - In 'gallery' mode, smtm now updates chart windows along with the 
      normal data allowing for automatically updated intra-daily or
      intra-weekly charts (longer horizons are updated less frequently
      than daily at Yahoo!). Gallery mode can be launched either via
      the command-line option '--gallery', or via the Chart->Gallery
      menu entry.
  
    - Small updates to documentation

 -- Dirk Eddelbuettel <edd@debian.org>  Tue,  6 Jan 2004 20:38:16 -0600

smtm (1.5.6) unstable; urgency=low

  * Bug fix release:
  
    - smtm: correct small error where defensive assignment of number of
      share was not used
    
    - smtm: use pack(-fill ...) throughtout, not pack(fill => ...), as this
      seems to trip up some 5.8.* versions of Perl. Thanks to Bruce Bowler
      for reporting this
  
    - smtm: use LWP::UserAgent->new(), not RequestAgent->new(), as this seems
      to trip up some ActiveState version. Thanks to Chris Eidem and others
      for reporting this.

 -- Dirk Eddelbuettel <edd@debian.org>  Wed, 15 Oct 2003 20:47:06 -0500

smtm (1.5.5) unstable; urgency=low

  * Bug fix release:
  
    - smtm: Chart images are no longer saved to a tempfile, but rather 
      passed directly (in base64 encoding) to the Photo widget
    
    - smtm: The Photo widget is now deleted on <destroy> plugging a
      memory hole
  
    - Makefile.PL: Depend on MIME::Base64
  

Makefile.PL
#
# Makefile.PL for smtm
#
# smtm --- A global stock ticker for X11 and Windoze
#
# Copyright (C) 1999 - 2007  Dirk Eddelbuettel <edd@debian.org>
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, write to the Free Software
# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
# 

use ExtUtils::MakeMaker;

WriteMakefile(
	      'NAME'		=> 'smtm',
	      'VERSION_FROM'	=> 'smtm',
	      'PREREQ_PM'	=> { "Date::Manip"	   => 5.0,
				     "Finance::YahooQuote" => 0.2,
				     "LWP::UserAgent"	   => 1.23,
				     "MIME::Base64"	   => 2.12,
				     "HTML::Parser"	   => 2.20,
				     "Tk"		   => 800.015,
				     "Tk::PNG"		   => 2.005
				 },
	      ($] >= 5.005 ?
	       ('ABSTRACT'   	=> 'Stock quote monitor and portfolio tool',
		'AUTHOR'     	=> 'Dirk Eddelbuettel (edd@debian.org)') : ()),
	      EXE_FILES 	=> ['smtm'],
	      'dist'		=> { COMPRESS => 'gzip', 
				     SUFFIX => '.gz', },
);

